extension.neon

```diff
@@ -69,7 +69,7 @@ services:
 
     - Efabrica\PHPStanLatte\Analyser\FileAnalyserFactory
     - Efabrica\PHPStanLatte\Analyser\AnalysedTemplatesRegistry(%analysedPaths%, %latte.reportUnanalysedTemplates%)
-    - Efabrica\PHPStanLatte\Compiler\LatteToPhpCompiler(%latte.tmpDir%)
+    - Efabrica\PHPStanLatte\Compiler\LatteToPhpCompiler(%latte.tmpDir%, ::md5(::json_encode(%latte%)))
```

Comment:
Do we need md5 here? In php code there is second one.

Reply:
I pass cachekey as string and i do not want to pass big string when it is not necessary

Reply:
I don’t know @spaze is an expert here but I think I read something about double hashing as not good practise. Just not sure if it is more possible conflicts then

Reply:
Double hashing is problem if you want to hide plaintext because counter intuitive it can make reverting it easier.

Bit for cahe it is imho ok. And also sexond hashong uses diffetent input

Reply:
🤷

Reply:
Doing md5(md5($foo)) wouldn't be ideal security-wise but in this case it's probably fine. Due to copy-on-write it may actually consume more memory than when passing just %latte% and doing the serialization when needed.

Reply:
I need to remember passed value